src/Elastic.Extensions.Logging.Common/LogEventBuilderExtensions.cs

```diff
@@ -0,0 +1,212 @@
+using System.Collections;
+using System.Text;
+using Elastic.CommonSchema;
+using Microsoft.Extensions.Logging;
+
+namespace Elastic.Extensions.Logging.Common;
+
+/// <summary> </summary>
+public static class LogEventBuilderExtensions
+{
+	/// <summary> </summary>
+	public static void AddScopeValues(this LogEvent logEvent, IExternalScopeProvider? scopeProvider, ILogEventCreationOptions options)
+	{
+		void AddScopeValue<TLocalState>(TLocalState scope, LogEvent log)
+		{
+			if (scope is null) return;
+
+			log.Labels ??= new Labels();
+			log.Scopes ??= new List<string>();
+
+
+			var scopeValues = (scope as IEnumerable<KeyValuePair<string, object>>)?.ToList();
```

Comment:
I think this is fine for now, but maybe we should consider a lower allocation approach with a rented array for these or even `stackalloc` for low volumes?

Reply:
I think we should but this is already in place today, only lights up in the PR because we moved it to a new shared library. 

Shall we follow up with that in another PR?